// hc-lib/hc.module.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { NgModule, } from '@angular/core';
import { HcDirective } from './hc.directive';
import { HcService } from './hc.service';
import { HTTP_INTERCEPTORS, HttpClientModule } from '@angular/common/http';
import { PostInterceptor } from "./interceptor/post.interceptor";
var HcModule = /** @class */ (function () {
    function HcModule() {
    }
    HcModule = __decorate([
        NgModule({
            imports: [HttpClientModule],
            exports: [HcDirective],
            declarations: [HcDirective],
            providers: [HcService, {
                    provide: HTTP_INTERCEPTORS,
                    useClass: PostInterceptor,
                    multi: true
                }],
        })
    ], HcModule);
    return HcModule;
}());
export { HcModule };
